src/cargo.js

```diff
@@ -31,7 +31,18 @@ export const cargoCommand = (target, release, subcmd = []) => {
   return cmd.join(' ');
 };
 
-export const handleCargo = async function (self, result) {
+export const handleCargo = async (self, result) => {
+  // result seems to not have a code, when the compilations succeeds, so we'll
+  // have to check existence first
+  if (result.code && result.code !== 0) {
+    throw makeCargoError(result);
```

Comment:
Could we use the already existing `BuildError` instead here to get the special error handling in `index.js`?

Reply:
Absolutely.